consul/src/main/scala/io/buoyant/consul/v1/KvApi.scala

```diff
@@ -95,15 +98,27 @@ class KvApi(
     path: String,
     datacenter: Option[String] = None,
     cas: Option[String] = None,
+    recurse: Boolean = false,
     retry: Boolean = false
   ): Future[Boolean] = {
     val req = mkreq(
       http.Method.Delete,
       s"$kvPrefix$path",
       "cas" -> cas,
+      "recurse" -> (if (recurse) Some(recurse.toString) else None),
       "dc" -> datacenter
     )
-    executeJson[Boolean](req, retry).map(_.value)
+    execute[Boolean](req, retry).map(_.value)
   }
+}
+
+object Key {
+  def mk(key: String, value: String): Key = Key(Some(key), Some(Base64.getEncoder.encodeToString(value.getBytes)))
+}
 
+case class Key(
+  Key: Option[String],
```

Comment:
I wonder whether I can drop `Option` from those fields as they 'must be set at all times'?